src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test.Common/CSharpTestLspServer.cs

```diff
@@ -129,6 +131,52 @@ public async ValueTask DisposeAsync()
             await _serverMessageHandler.DisposeAsync();
         }
 
+        #region Document Change Methods
+
+        public async Task OpenDocumentAsync(Uri documentUri, string documentText)
```

Comment:
`OpenDocumentAsync` and `ReplaceTextAsync` were added due to provisional completion. Since provisional completion modifies the C# backing document, we had to communicate this to the C# LSP server which involves making the backing document a tracked document via `textDocument/didOpen`.